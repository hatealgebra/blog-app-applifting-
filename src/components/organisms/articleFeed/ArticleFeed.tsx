import React from 'react';
import { TCompleteArticles } from 'customTypes/schema';

import { Components } from '@customTypes/declarations';
import ArticlePreview from '../../molecules/articlePreview/ArticlePreview';
import {
  ArticlesContainer,
  CenterContainer,
  StyledArticleFeed,
} from './articleFeed.styled';

interface ArticleFeedProps {
  items: TCompleteArticles;
}

const ArticleFeed = ({ items }: ArticleFeedProps) => {
  return (
    <StyledArticleFeed className="article-feed">
      <h1 className="article-feed__heading">Recent articles</h1>
      <div className="article-feed__articles">
        {items === undefined || !items.length ? (
          <CenterContainer>
            <h3>No Articles to show</h3>
          </CenterContainer>
        ) : (
          <ArticlesContainer>
            {items.map(
              ({
                articleId,
                title,
                perex,
                createdAt,
                imageId,
                imageBase64,
                comments,
              }: Components['schemas']['Article'] &
                Components['schemas']['ArticleDetail'] & {
                  imageBase64: 'string';
                }) => (
                <ArticlePreview
                  key={articleId}
                  articleId={articleId}
                  title={title}
                  perex={perex}
                  createdAt={createdAt}
                  imageId={imageId}
                  comments={comments}
                  imageBase64={imageBase64}
                  author="Unknow author"
                />
              )
            )}
          </ArticlesContainer>
        )}
      </div>
    </StyledArticleFeed>
  );
};

export default ArticleFeed;
