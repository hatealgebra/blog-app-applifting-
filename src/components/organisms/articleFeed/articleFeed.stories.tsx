import React from 'react';
import ArticleFeed from './ArticleFeed';
import allArticlesDetailMockJSON from '../../../__mocks__/asyncData/get/articlesDetailsResponse.mock.json';

export const ArticleFeedLoading = () => (
  <ArticleFeed items={allArticlesDetailMockJSON.items} />
);

export const ArticleFeedEmpty = () => <ArticleFeed items={[]} />;

export default {
  title: 'Organisms/Article Feed',
  component: ArticleFeed,
};
