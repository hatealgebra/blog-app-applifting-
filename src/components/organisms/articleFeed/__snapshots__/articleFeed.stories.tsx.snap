// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Article Feed ArticleFeedEmpty smoke-test 1`] = `
<div class="sc-kMbQoj iDNxwp article-feed">
  <h1 class="article-feed__heading">
    Recent articles
  </h1>
  <div class="article-feed__articles">
    <div class="sc-bYUneI dWksIS">
      <h3>
        No Articles to show
      </h3>
    </div>
  </div>
</div>
`;

exports[`Organisms/Article Feed ArticleFeedLoading smoke-test 1`] = `
<div class="sc-kMbQoj iDNxwp article-feed">
  <h1 class="article-feed__heading">
    Recent articles
  </h1>
  <div class="article-feed__articles">
    <div class="sc-ezGUZh bBMuWw">
      <article class="sc-eLSjS kPKThh article-preview">
        <img class="sc-bqOBKd FGsFx article-preview__img"
             src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAEACAIAAADTED8xAAADMElEQVR4nOzVwQnAIBQFQYXff81RUkQCOyDj1YOPnbXWPmeTRef+/3O/OyBjzh3CD95BfqICMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMO0TAAD//2Anhf4QtqobAAAAAElFTkSuQmCC"
             alt="British Cats: Charming Companions of the Isles preview image"
        >
        <h3 class="article-preview__title">
          British Cats: Charming Companions of the Isles
        </h3>
        <div class="sc-ksJisA icZPyt article-preview__row-one">
          <span>
            Unknow author
          </span>
          <svg stroke="currentColor"
               fill="currentColor"
               stroke-width="0"
               viewbox="0 0 8 16"
               height="1em"
               width="1em"
               xmlns="http://www.w3.org/2000/svg"
          >
            <path fill-rule="evenodd"
                  d="M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z"
            >
            </path>
          </svg>
          <span>
            07/02/23
          </span>
        </div>
        <p class="article-preview__text">
          British cats, known for their distinctive charm and character, hold a special place in the hearts of cat enthusiasts around the wo...
        </p>
        <div class="sc-ksJisA icZPyt article-preview__row-two">
          <a class="sc-jrkPcm eEbEbs"
             href="/articles/c7b1f38a-e70f-4795-bdd8-802c68eb32dc"
          >
            Read whole article
          </a>
          <span>
            4 comments
          </span>
        </div>
      </article>
      <article class="sc-eLSjS kPKThh article-preview">
        <img class="sc-bqOBKd FGsFx article-preview__img"
             src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAEACAIAAADTED8xAAADMElEQVR4nOzVwQnAIBQFQYXff81RUkQCOyDj1YOPnbXWPmeTRef+/3O/OyBjzh3CD95BfqICMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMO0TAAD//2Anhf4QtqobAAAAAElFTkSuQmCC"
             alt="Bean the Cat: A Tale of Adventure, Friendship, and Whiskers preview image"
        >
        <h3 class="article-preview__title">
          Bean the Cat: A Tale of Adventure, Friendship, and...
        </h3>
        <div class="sc-ksJisA icZPyt article-preview__row-one">
          <span>
            Unknow author
          </span>
          <svg stroke="currentColor"
               fill="currentColor"
               stroke-width="0"
               viewbox="0 0 8 16"
               height="1em"
               width="1em"
               xmlns="http://www.w3.org/2000/svg"
          >
            <path fill-rule="evenodd"
                  d="M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z"
            >
            </path>
          </svg>
          <span>
            07/02/23
          </span>
        </div>
        <p class="article-preview__text">
          In the heart of a quaint little village nestled between rolling hills, a remarkable feline named Bean resided. Bean wasn't your ty...
        </p>
        <div class="sc-ksJisA icZPyt article-preview__row-two">
          <a class="sc-jrkPcm eEbEbs"
             href="/articles/d1cfbb7d-e0d9-4341-a131-f1dd077a6fb0"
          >
            Read whole article
          </a>
          <span>
            4 comments
          </span>
        </div>
      </article>
      <article class="sc-eLSjS kPKThh article-preview">
        <img class="sc-bqOBKd FGsFx article-preview__img"
             src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAEACAIAAADTED8xAAADMElEQVR4nOzVwQnAIBQFQYXff81RUkQCOyDj1YOPnbXWPmeTRef+/3O/OyBjzh3CD95BfqICMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMO0TAAD//2Anhf4QtqobAAAAAElFTkSuQmCC"
             alt="The Irresistible Charm of Orange Cats: Exploring Their Purr-sonalities and Memes preview image"
        >
        <h3 class="article-preview__title">
          The Irresistible Charm of Orange Cats: Exploring T...
        </h3>
        <div class="sc-ksJisA icZPyt article-preview__row-one">
          <span>
            Unknow author
          </span>
          <svg stroke="currentColor"
               fill="currentColor"
               stroke-width="0"
               viewbox="0 0 8 16"
               height="1em"
               width="1em"
               xmlns="http://www.w3.org/2000/svg"
          >
            <path fill-rule="evenodd"
                  d="M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z"
            >
            </path>
          </svg>
          <span>
            07/02/23
          </span>
        </div>
        <p class="article-preview__text">
          Orange cats, with their vibrant coats and captivating personalities, have captured the hearts of cat lovers around the world. Thes...
        </p>
        <div class="sc-ksJisA icZPyt article-preview__row-two">
          <a class="sc-jrkPcm eEbEbs"
             href="/articles/abd4dcae-d68c-4aa2-9651-e971d4389425"
          >
            Read whole article
          </a>
          <span>
            4 comments
          </span>
        </div>
      </article>
      <article class="sc-eLSjS kPKThh article-preview">
        <img class="sc-bqOBKd FGsFx article-preview__img"
             src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAEACAIAAADTED8xAAADMElEQVR4nOzVwQnAIBQFQYXff81RUkQCOyDj1YOPnbXWPmeTRef+/3O/OyBjzh3CD95BfqICMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMO0TAAD//2Anhf4QtqobAAAAAElFTkSuQmCC"
             alt="Unraveling the Mystery: The Enigmatic World of Grumpy Cats and Their Peculiar Demeanor preview image"
        >
        <h3 class="article-preview__title">
          Unraveling the Mystery: The Enigmatic World of Gru...
        </h3>
        <div class="sc-ksJisA icZPyt article-preview__row-one">
          <span>
            Unknow author
          </span>
          <svg stroke="currentColor"
               fill="currentColor"
               stroke-width="0"
               viewbox="0 0 8 16"
               height="1em"
               width="1em"
               xmlns="http://www.w3.org/2000/svg"
          >
            <path fill-rule="evenodd"
                  d="M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z"
            >
            </path>
          </svg>
          <span>
            07/02/23
          </span>
        </div>
        <p class="article-preview__text">
          In the realm of feline companionship, there exist an intriguing subset of cats that have garnered attention not for their charming...
        </p>
        <div class="sc-ksJisA icZPyt article-preview__row-two">
          <a class="sc-jrkPcm eEbEbs"
             href="/articles/541d7c47-c7ec-4ec9-a2b0-b6a817278c40"
          >
            Read whole article
          </a>
          <span>
            4 comments
          </span>
        </div>
      </article>
      <article class="sc-eLSjS kPKThh article-preview">
        <img class="sc-bqOBKd FGsFx article-preview__img"
             src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAEACAIAAADTED8xAAADMElEQVR4nOzVwQnAIBQFQYXff81RUkQCOyDj1YOPnbXWPmeTRef+/3O/OyBjzh3CD95BfqICMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMO0TAAD//2Anhf4QtqobAAAAAElFTkSuQmCC"
             alt="The Fascinating World of Cat Sleeping Habits: Exploring Feline Slumber preview image"
        >
        <h3 class="article-preview__title">
          The Fascinating World of Cat Sleeping Habits: Expl...
        </h3>
        <div class="sc-ksJisA icZPyt article-preview__row-one">
          <span>
            Unknow author
          </span>
          <svg stroke="currentColor"
               fill="currentColor"
               stroke-width="0"
               viewbox="0 0 8 16"
               height="1em"
               width="1em"
               xmlns="http://www.w3.org/2000/svg"
          >
            <path fill-rule="evenodd"
                  d="M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z"
            >
            </path>
          </svg>
          <span>
            07/03/23
          </span>
        </div>
        <p class="article-preview__text">
          Cats are renowned for their unique and often mysterious behaviors, and one of the most intriguing aspects of their lives is their ...
        </p>
        <div class="sc-ksJisA icZPyt article-preview__row-two">
          <a class="sc-jrkPcm eEbEbs"
             href="/articles/2491852b-6566-4fae-ace3-4510cd40e81a"
          >
            Read whole article
          </a>
          <span>
            4 comments
          </span>
        </div>
      </article>
    </div>
  </div>
</div>
`;
